front: drop tsv2 switch

SharglutDev opened the pull request:
Operational studies and stdcm both now use tsv2 model by default

e2e tests have been adapted and some removed (see 2nd commit) in accordance with @Maymanaf 

close #8059

SharglutDev committed:
front: drop tsv2 switch

Operational studies and stdcm both now use tsv2 model by default

Uriel-Sautron commented on front/src/modules/trainschedule/components/ManageTrainSchedule/TrainSettings.tsx:
```diff
@@ -97,40 +95,22 @@ export default function TrainSettings() {
+          label={
+            <>
+              <MdOutlineAccessTime />
+              {/* TOSO TS2 : rename trainScheduleDepartureTime key to trainScheduleStartTime everywhere */}
```
```suggestion
              {/* TODO TS2 : rename trainScheduleDepartureTime key to trainScheduleStartTime everywhere */}
```

Uriel-Sautron replied:
Typo

SharglutDev replied:
Done

Uriel-Sautron reviewed (commented):
Review and tested on operational studie and stdcm no problem for me.
I'll keep testing.

SharglutDev committed:
front: adapt e2e tests to use tsv2

- e2e tests are now using tsv2 model
- remove allowances e2e tests as they are deprecated
- rolling stock comfort type properly uses tsv2 model

theocrsb reviewed (approved):
LGTM and tested

clarani commented on front/src/applications/stdcm/hooks/useStdcm.ts:
```diff
@@ -199,11 +109,7 @@ const useStdcm = () => {
 
   const launchStdcmRequest = async () => {
```
You can even rename launchStdcmRequestV2 by launchStdcmRequest and merge the 2 functions

SharglutDev replied:
Done

clarani commented on front/src/common/UserSettings.tsx:
```diff
@@ -89,21 +80,7 @@ const UserSettings = () => {
-              name="train-schedule-version-switch"
-              onChange={() => {
-                dispatch(switchTrainScheduleV2Activated());
-                resetStore();
```
I think you can remove the function resetStore, we should not call it when toggling the stdcm switch

SharglutDev replied:
Done

clarani reviewed (commented):
The code LGTM, I still need to test:
- [x] Eex
- [x] stdcm
- [x] rolling stock editor

clarani reviewed (commented):
Code looks good to me, only found 2 bugs:
- when I select a rolling stock with heating or air conditioning and reopen the rolling stock selector, then the selected rolling stock is correct but has not the correct mode (it has standard)
- when editing a rolling stock, in the effort curves tab, there is a missing translation for the air conditioning tag

SharglutDev commented:
Should be fixed :)

clarani reviewed (approved):
LGTM, tested :white_check_mark:

clarani reviewed (commented):
Last bug: if I duplicate a rollingStock (tgv 2N2 for instance) and I edit it, there is a missing translation:
![image](https://github.com/user-attachments/assets/192c91c3-7f89-4236-9361-007e439b7314)
